experimental/packages/opentelemetry-instrumentation-fetch/test/fetch.test.ts

```diff
@@ -622,6 +631,66 @@ describe('fetch', () => {
     });
   });
 
+  describe.only('when response has url', () => {
+    beforeEach(async () => {
+      const propagateTraceHeaderCorsUrls = [url];
+      await prepareData(redirectedUrl, { propagateTraceHeaderCorsUrls });
+    });
+
+    afterEach(() => {
+      clearData();
+    });
+
+    it('span should have correct attributes', () => {
+      const span: tracing.ReadableSpan = exportSpy.args[1][0][0];
+      const attributes = span.attributes;
+      const keys = Object.keys(attributes);
+
+      assert.ok(
+        attributes[keys[0]] !== '',
+        `attributes ${AttributeNames.COMPONENT} is not defined`
+      );
+      assert.strictEqual(
+        attributes[keys[1]],
+        'GET',
+        `attributes ${SemanticAttributes.HTTP_METHOD} is wrong`
+      );
+      assert.strictEqual(
+        attributes[keys[2]],
+        redirectedUrl,
```

Comment:
Does it make sense that the `http.url` attribute's value is the url before the redirection, but `http.host` is the host after the redirection?

Reply:
Hmm, you're right, it does indeed feel strange. As far as I can tell, it also does not follow the [spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/http.md#http-request-retries-and-redirects). There it states that:
> Retries and redirects cause more than one physical HTTP request to be sent. A CLIENT span SHOULD be created for each one of these physical requests. No span is created corresponding to the "logical" (encompassing) request.

It looks like to me that we'd need to instrument fetch in a way that it creates spans on each redirect (in that case `host.host` and `http.url` should be the same). However, I have to admit that I do not know enough about the internals of `fetch` right now to come up with a way to implement that. We should create an issue to track this, though.

Reply:
Since the discrepancy is introduced by this change, a better fix for now should be to just remove `spanResponse` variable, and then implement proper redirection spans.

Reply:
Sounds good. If there's a way to create proper redirection spans we should do it :+1:

Reply:
Doesn't look like it is possible, sadly.
Redirects are not exposed to APIs: https://fetch.spec.whatwg.org/#atomic-http-redirect-handling
There's a `manual` redirect mode, but it can't be used in any useful way. 
> `"manual"`
> Retrieves an [opaque-redirect filtered response](https://fetch.spec.whatwg.org/#concept-filtered-response-opaque-redirect) when a request is met with a redirect, to allow a service worker to replay the redirect offline. The response is otherwise indistinguishable from a [network error](https://fetch.spec.whatwg.org/#concept-network-error), to not violate [atomic HTTP redirect handling](https://fetch.spec.whatwg.org/#atomic-http-redirect-handling).

> An opaque-redirect filtered response is a [filtered response](https://fetch.spec.whatwg.org/#concept-filtered-response) whose [type](https://fetch.spec.whatwg.org/#concept-response-type) is "opaqueredirect", [status](https://fetch.spec.whatwg.org/#concept-response-status) is 0, [status message](https://fetch.spec.whatwg.org/#concept-response-status-message) is the empty byte sequence, [header list](https://fetch.spec.whatwg.org/#concept-response-header-list) is empty, and [body](https://fetch.spec.whatwg.org/#concept-response-body) is null.